### PR TITLE
linters: semgrep rule to replace AWSR002

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #18721 
Preceded by #7926

Output from semgrep testing:

```
$ semgrep
ran 21 rules on 2451 files: 0 findings
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Running on "current" (4b72ef50b3c1fbf120f3093766b5cf35541f1f95) `main`:

```console
% semgrep --config .semgrep.yml                     
running 20 rules...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|20/20
aws/resource_aws_elastic_beanstalk_application.go
severity:warning rule:helper-schema-ResourceData-Set-tags: (schema.ResourceData).Set() call with the tags key should include IgnoreConfig in the value
266:	tags = tags.IgnoreElasticbeanstalk().IgnoreConfig(ignoreTagsConfig)
267:
268:	//lintignore:AWSR002
269:	if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
270:		return fmt.Errorf("error setting tags: %w", err)
271:	}

aws/resource_aws_elastic_beanstalk_application_version.go
severity:warning rule:helper-schema-ResourceData-Set-tags: (schema.ResourceData).Set() call with the tags key should include IgnoreConfig in the value
134:	tags = tags.IgnoreElasticbeanstalk().IgnoreConfig(ignoreTagsConfig)
135:
136:	//lintignore:AWSR002
137:	if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
138:		return fmt.Errorf("error setting tags: %w", err)
139:	}

aws/resource_aws_elastic_beanstalk_environment.go
severity:warning rule:helper-schema-ResourceData-Set-tags: (schema.ResourceData).Set() call with the tags key should include IgnoreConfig in the value
627:	tags = tags.IgnoreElasticbeanstalk().IgnoreConfig(ignoreTagsConfig)
628:
629:	//lintignore:AWSR002
630:	if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
631:		return fmt.Errorf("error setting tags: %w", err)
632:	}

aws/resource_aws_waf_rate_based_rule.go
severity:warning rule:helper-schema-ResourceData-Set-tags: (schema.ResourceData).Set() call with the tags key should include IgnoreConfig in the value
164:	tags := tagList.IgnoreAws().IgnoreConfig(ignoreTagsConfig)
165:
166:	//lintignore:AWSR002
167:	if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
168:		return fmt.Errorf("error setting tags: %w", err)
169:	}

aws/resource_aws_wafregional_rate_based_rule.go
severity:warning rule:helper-schema-ResourceData-Set-tags: (schema.ResourceData).Set() call with the tags key should include IgnoreConfig in the value
164:	tags := tagList.IgnoreAws().IgnoreConfig(ignoreTagsConfig)
165:
166:	//lintignore:AWSR002
167:	if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
168:		return fmt.Errorf("error setting tags: %w", err)
169:	}
ran 20 rules on 2451 files: 5 findings
```

**Comment by anGie44** [comment]
> Running on "current" ([4b72ef5](https://github.com/hashicorp/terraform-provider-aws/commit/4b72ef50b3c1fbf120f3093766b5cf35541f1f95)) `main`:

Ohh interesting! I'll add a pattern for the `IgnoreElasticBeanstalk` and generalize the `tags.IgnoreAws()` so `tagsList` gets included. I'm not too sure if the entire rule though is optimal 😅 a lot of `patter-not` conditions, I wonder if there's a better way to group them 🤔 ?

File: .semgrep.yml [hunk 1]
```diff
@@ -268,7 +268,7 @@ rules:
       include:
         - aws/
     patterns:
-      - pattern-inside: func $READMETHOD(...) $ERRORTYPE { ... }
+      - pattern-inside: func $FUNCNAME(...) $ERRORTYPE { ... }
```
Comment (anGie44) [inline .semgrep.yml:271]: just changed to reflect we're also looking inside `dataSource..Read` and `flatten..Resource` methods as well

Comment (anGie44) [reply .semgrep.yml:271]: on second thought, the rule misses looking at the data-source logic 😮

**Review by YakDriver — APPROVED** [review]
Excellent! 🎉 

```
% semgrep
running 22 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|22/22
ran 22 rules on 2459 files: 0 findings
```

**Pull request merged**